### DISCUSSION TITLE
Typo Tolerance 'min' and Menu item with static refinements

### DISCUSSION THREAD

**Discussion by AntoineDuComptoirDesPharmacies** [algolia/instantsearch#5894] [Q&A]
Hi,

We encounter a strange behavior and we are wondering how to fix it.
Context : 
Imagine you have an index with three items : 
```
{
   id: 1,
   category: "Pants",
   description: "BLUE"
}
{
   id: 2,
   category: "Pants"
   description: "BLUA"
}
{
   id: 3,
   category: "T-Shirt"
   description: "BLUA"
}
```

We have a static menu with the potential values : 
`"Any category", "Pants", "T-Shirt"`

Our Typo tolerance is set to `min`.

When arriving on the page, we have the following count : 
```
"Any category": 3 <= This is the current selected
"Pants" : 2
"T-Shirt": 1
```

When we search for query "BLUE", the count changes and become this, following typo tolerance min : 
```
"Any category": 1 <= This is the current selected
"Pants" : 1
"T-Shirt": 0
```

When we click on "T-Shirt", the count change again as the main query do not have anymore the results of "Pants" so typo tolerance change the result of "T-Shirt" for the same query. New count is : 
```
"Any category": 1
"Pants" : 1
"T-Shirt": 1 <= This is the current selected
```

Do you have any genius idea in order to get the following final state, something like asking "T-Shirt" query to take into account the potential typo of "Any category" : 
```
"Any category": 1
"Pants" : 1
"T-Shirt": 0 <= This is the current selected
```

Note : We can not filters hits after search result because we are using pagination of Algolia.

Thanks in advance for your enlightments !
Yours faithfully,
LCDP

**Comment by Haroenv** [comment]
I don't think facet count is related to typo tolerance, but what I wonder is if the count actually is exhaustive, you can see that with the `exhaustiveFacetCount` key in the network response

**Reply by AntoineDuComptoirDesPharmacies** [reply]
Thank you for your response @Haroenv.

From what we can see, the FacetCount take into account the typo tolerance.
Here is a result with typo tolerance 'min'
![image](https://github.com/algolia/instantsearch/assets/104082030/b0f7cd5b-7898-4940-8ae4-7a691e42d05e)

Here is a result with typo tolerance 'true' (same query)
![image](https://github.com/algolia/instantsearch/assets/104082030/c1372780-d67e-4548-ad60-b376303f0569)

The problem exists when you are applying a facet value that will match results with typo, and if without facet filtering there is match without typo. In this case, you will get a different facet count depending if you apply this facet or not.
This can happen if : 
- You display the facet even if the count is zero
- You sync the instant search state with URL and user come back to this URL after some index items changes (for example an element with 0 typo get deleted, and now the new result have typo for this facet)

On an other hand, if the facet count do not match the typo, we would produce an other strange behavior. With the same example : 

When arriving on the page, we have the following count :
```
"Any category": 3 <= This is the current selected
"Pants" : 2
"T-Shirt": 1
```

When we search for query "BLUE", the count changes and become this, following typo tolerance min (with the count of any category being wrong) :
```
"Any category": 1 <= This is the current selected
"Pants" : 1
"T-Shirt": **1**
```

When we click on "T-Shirt", no change of count but "Any category" count is still wrong :
```
"Any category": 1
"Pants" : 1
"T-Shirt": 1 <= This is the current selected
```

Of course, in this case, the total count of "Any category" is the `nbHits` located in the `rawResults` associated with this facet.
We could not sum the facet counts because an item can have multiple value for a same facet.

Yous faithfully,
LCDP
